sycl/doc/extensions/experimental/sycl_ext_oneapi_annotated_ptr.asciidoc

```diff
@@ -532,7 +532,100 @@ class annotated_ref {
     T operator++(int) const;
     T operator--() const;
     T operator--(int) const;
+
+    // Overloaded arithmetic operators
```

Comment:
Things I consider problems and need to be changed:
- All overloads should be hidden friends.

Things I consider nicer style and would prefer:
- We should use the OP syntax to make this shorter.
- There is no need for the R template argument.
- No advantage of passing ann_ref by reference.
- We don't need 3 separate overloads

Suggestion:
```
 //get_T is exposition only 
 template <class O> struct get_T { using type = O; };
 template <class O, class PL> struct get_T<annotated_ref<O,PL>> { using type = O; };
 template <class O> using get_T_t = typename get_T<O>::type;
 template <class T1, class T2>
 friend decltype(std::declval<get_T_t<T1>>() OP std::declval<get_T_t<T1>>()) operatorOP(T1, T2);
```

Reply:
I think we also need the unary arithmetic, logical, and comparison operators. Unary is also needed because otherwise:

```
struct S {
    operator int();
};
void f(annotated_ref<S> r) { +r; }
```
doesn't work. I see no reason why we should treat logical and comparison different from arithmetic.

Suggestion for unary operators:
```
auto operator OP() -> decltype(OP std::declval<T>());
```

Reply:
Hi.
> We should use the OP syntax to make this shorter.
What do you mean by using OP syntax?
> No advantage of passing ann_ref by reference.
annotated_ref copy constructor is deleted so we have to pass by reference.

Reply:
Updated the PR